website/markdown/docs/usage/adoption-guidelines.mdx

```diff
@@ -0,0 +1,89 @@
+How tedious this process is,
+depends on how complex your projects are.
+
+As you probably know,
+Xcode doesn't prevent developers from adding accidental complexity.
+That leads to complexity in the shape of _groups that don't match the directory structure_,
+_files that are shared across targets_, or
+_file references that point to nonexisting files_ _(to mention some)._
+All that accumulated complexity makes it hard for us to provide a command that reliably ports migrates project.
+
+Moreover, manual migration is an **excellent exercise to clean up and simplify** your projects.
+Not only the developers in your project will be thankful for that,
+but Xcode,
+who will be faster processing and indexing them.
+Once you have fully adopted Tuist,
+it will make sure that projects are consistently defined and that they remain simple.
+
+In the aim of easing that work,
+we are giving you some guidelines based on the feedback that we have received from the users.
+
+<Message
+  info
+  title="Migration commands"
+  description="We are currently working on providing commands to assist you in the process. Until those commands are available, you might consider building simple CLI tools that leverage [XcodeProj](https://github.com/tuist/xcodeproj)"
+/>
+
+## Guidelines
+
+#### Extract your build settings into `.xcconfig` files
+
+The leaner the projects, the easier they are to migrate.
+To make them thinner,
+you can extract information like build settings that can be defined outside Xcode projects.
+You must prevent developers from defining settings using build settings.
+You can do that by implementing a script that reads the build settings using [XcodeProj](https://github.com/tuist/xcodeproj),
+and fails if the settings of projects and targets are not empty.
+
+#### Migrate the most dependant projects first
+
+Those are usually simpler and contain fewer dependencies than the rest.
+That makes them good candidates from which we can start the migration.
+
+#### Remove broken references
+
+Go through your target sources and resources build phases, and delete references to files that are missing.
+Although Xcode's build system ignores them, they might complicate the migration.
+
+#### Remove files that are not used by any target
+
+There might be files that were once part of the project,
+but that they are no longer needed.
+Find and remove them.
+Otherwise,
+you might run into compilation issues because your glob patterns _(e.g. `**/*.swift`)_ ended up matching them,
+and adding them as sources of your targets.
+
+#### Prevent modifications of Xcode projects after they've been migrated
+
+After the migration of each project,
+add a script that fails CI if someone modifies the Xcode project directly,
+and lets developers know that they should change the manifest files instead.
+
+You can use a tool like [xcdiff](https://github.com/bloomberg/xcdiff) for that.
```

Comment:
if you are still checking in your pbxprojs on git, you can also go with a simple solution:
1) generate the project on CI
2) compare it with the project committed in the PR
if they are not the same file, fail the PR